test.py

```diff
@@ -1,131 +1,123 @@
-            and func_exists(mod_path, "sliding_dot_product")
-            and func_exists(mod_path, "setup")
-        ):
-            mod_name = f"sdp.{m[1]}"
-            mod = importlib.import_module(mod_name)
-            mods.append(mod)
-
-    return mods
-
-
-if __name__ == "__main__":
-    parser = argparse.ArgumentParser(
-        description="./test.py -noheader -pmin 6 -pmax 23 -pdiff 3 pyfftw challenger"
-    )
-    parser.add_argument("-noheader", default=False, action="store_true")
-    parser.add_argument("-timeout", default=5.0, type=float, help="Number of seconds to wait for a run before timing out")
-    parser.add_argument("-pequal", default=False, action="store_true", help="Compute `len(Q) == len(T)`")
-    parser.add_argument("-niter", default=4, type=int, help="Number of iterations to run")
-    parser.add_argument("-pmin", default=6, type=int, help="Minimum 2^p to use")
-    parser.add_argument("-pmax", default=27, type=int, help="Maximum 2^p to use")
-    parser.add_argument("-pdiff", default=100, type=int, help="Maximum deviation from the minimum 2^p allowed")
-    parser.add_argument("-ignore", default=None, nargs="*", help="Keyword of modules to match and ignore")
-    parser.add_argument("include", default=None, nargs="*", help="Keyword of modules to match and include")
-    args = parser.parse_args()
-
-    modules = import_sdp_mods(args.include, args.ignore)
-
-    noheader = args.noheader
-    timeout = args.timeout
-    if args.pequal:
-        skip_p_equal = 0
-    else:
-        skip_p_equal = 1
-    n_iter = args.niter
-    p_min = args.pmin
-    p_max = args.pmax
-    p_diff = args.pdiff
-
-    if not noheader:
-        print(f"module,len_Q,len_T,n_iter,time", flush=True)
-
-    start_timing = time.time()
+import pytest
+import utils
+
+from numpy import testing as npt
+from operator import eq, lt
+from scipy.fft import next_fast_len
+
+# README
+# Real FFT algorithm performs more efficiently when the length
+# of the input array `arr` is composed of small prime factors.
+# The next_fast_len(arr, real=True) function from Scipy returns
+# the same length if len(arr) is composed of a subset of
+# prime numbers 2, 3, 5. Therefore, these radices are
+# considered as the most efficient for the real FFT algorithm.
+
+# To ensure that the tests cover different cases, the following cases
+# are considered:
+# 1. len(T) is even, and len(T) == next_fast_len(len(T))
+# 2. len(T) is odd, and len(T) == next_fast_len(len(T))
+# 3. len(T) is even, and len(T) < next_fast_len(len(T))
+# 4. len(T) is odd, and len(T) < next_fast_len(len(T))
+# And 5. a special case of 1, where len(T) is power of 2.
+
+# Therefore:
+# 1. len(T) is composed of 2 and a subset of {3, 5}
+# 2. len(T) is composed of a subset of {3, 5}
+# 3. len(T) is composed of a subset of {7, 11, 13, ...} and 2
+# 4. len(T) is composed of a subset of {7, 11, 13, ...}
+# 5. len(T) is power of 2
+
+# In some cases, the prime factors are powered to a certain degree
```

Comment:
"are raised to a power of a certain degree"? I can understand your meaning but "powered" is somewhat overloaded

Reply:
Right! That was not a good passive sentence 😅 I think what you suggested, i.e. `raised`, is the common term.